tests/unit/test_session.py

```diff
@@ -45,7 +46,7 @@ def setUp(self):
         self.environ_patch = mock.patch('os.environ', self.environ)
         self.environ_patch.start()
         self.environ['FOO_PROFILE'] = 'foo'
-        self.environ['FOO_REGION'] = 'moon-west-1'
```

Comment:
I'm working right now on making it so that we can guess the endpoint if we're unable to resolve an endpoint. This would allow us to keep the existing behavior of guessing as a last resort.